src/syntax_highlight.rs

```diff
@@ -34,7 +36,8 @@ unsafe impl Sync for Setup {}
 lazy_static!{
     static ref SETUP: Setup = Setup {
         syntax_set: SyntaxSet::load_defaults_newlines(),
-        theme_set: ThemeSet::load_defaults()
+        theme_set: ThemeSet::load_defaults(),
+        theme_name: RefCell::new(DEFAULT_THEME_NAME.to_owned()),
```

Comment:
:grin:

My pros (:+1:) and cons (:-1:) are pretty subjective.

The alternate patch set...

1.  :+1: Avoids using a global variable `SETUP.theme_name` in favor of scoped variable.
2.  :+1: Simplifies the PR by dropping two patches.
4.  :+1: Puts `theme` in `Config` (I'm assuming this continues to keep all config file options within one struct).
5.  :neutral_face: Wraps `initialize_codeblock()` in a closure.
6.  :neutral_face: No `RefCell`.  I'm not sure why this would matter.

The original patch set...

1.  :+1: Keeps `theme_name` organized more closely with relevant code in `syntax_highlight.rs`.
2.  :-1: Uses the global variable mentioned above.
3.  :-1: Reverts quite a bit of @mre's initial work (which was heading in the direction I favored).
4.  :-1: Stores a config file option separate from `Config` struct.
5.  :neutral_face: Works now, but global vars tend to cause problems later.

My OCD prefers the alternate patch set.  I would have submitted it instead had I figured it out sooner.